### ISSUE TITLE
WYSIWYG-markdown hybrid

### ISSUE THREAD

**Issue by SamPaplauskas** [StirlingCodingClub/studyGroup#12]
Not technically git related but I think we will be coming across Rmarkdown in coding club soon...

So I have been wondering:
1. Has anybody come across a WYSIWYG-markdown hybrid?
2. Would this even be practical?

In anticipation of a response to 2, I imagine that it is probably accessible enough to learn how to write markdown code, AND the different iterations and updates of markdown that people use would complicate the process of maintaining this.

**Comment by bradduthie** [comment]
There is a kind of hybrid [available online](https://dillinger.io/), though this is specifically limited to markdown, not Rmarkdown (though this might not matter much for getting started -- Rmarkdown just adds some nice features to integrate R). It might not be practical to write in the browser like this in the long term (one nice thing about markdown is that it allows you to focus just on the content and not worry about how it is formatted until later), but it could be a really useful way to get a feel for what markdown is doing. You can even save to to GitHub and Dropbox, or export as a PDF or HTML.

Future Plans
=======================================================

In the near future (i.e., sometime before early November), I'd be happy to do a Wednesday seminar on how to write a manuscript in Rmarkdown (this seems to be winning a lot of votes in the [list of potential topics](https://github.com/StirlingCodingClub/studyGroup/issues/6)), including the basics of Markdown more generally (note that comments in these issues are written in Markdown, like the numbered list you wrote above or the titles and links that I've used here -- click on the '...' and 'Edit' on anyone's comments to see the underlying code).

In the mean time, you can play with Rmarkdown a bit with [the Rmd file](https://github.com/StirlingCodingClub/version_control/blob/master/vc_notes.Rmd) that produced this week's [notes on version control](https://stirlingcodingclub.github.io/version_control/vc_notes.html) (click on the 'raw' button to see the [underlying code](https://raw.githubusercontent.com/StirlingCodingClub/version_control/master/vc_notes.Rmd) -- you can also paste this code directly into Rmarkdown, then click the 'Knit' button to see it build the HTML).

**Comment by SamPaplauskas** [comment]
That is cool - Dillinger gets a thumbs up for me

My initial query was more along the lines of - is there a user interface for markdown?

This is out of curiosity rather than needing a dumbed down version

**Comment by bradduthie** [comment]
@SamPaplauskas Ah, I see what you mean. I'm actually not sure if there's an especially good user interface for markdown. The [gedit](https://wiki.gnome.org/Apps/Gedit) text editor I use seems to recognise and do a good job of highlighting files saved with a "md" or "mkd" extension, but I'm sure there are better ones with a proper UI. Rstudio is of course good for Rmarkdown, but doesn't seem to highlight things as well for "md" and "mkd" (as opposed to "Rmd").

**Comment by SamPaplauskas** [comment]
Yes I agree - the highlighting etc of code is super useful - I just find it interesting to think about what a UI could do for the process
